hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractSeek.java

```diff
@@ -143,7 +143,8 @@ protected AbstractFSContract createContract(Configuration conf) {
   public void teardown() throws Exception {
     super.teardown();
     S3AFileSystem fs = getFileSystem();
-    if (fs.getConf().getBoolean(FS_S3A_IMPL_DISABLE_CACHE, false)) {
+    if (fs != null && fs.getConf().getBoolean(FS_S3A_IMPL_DISABLE_CACHE,
```

Comment:
Are these null check required?

Reply:
they are. IF test setup() fails then fs may be null, teardown will be called and you'll see the error in teardown (here an NPE) rather than whatever caused setup failure.